cassandra/cluster.py

```diff
@@ -1044,14 +1045,26 @@ def _ensure_core_connections(self):
             for pool in session._pools.values():
                 pool.ensure_core_connections()
 
-    def submit_schema_refresh(self, keyspace=None, table=None):
+    def refresh_schema(self, keyspace=None, table=None, usertype=None, schema_agreement_wait=None):
+        """
+        Synchronously refresh the schema metadata.
+        By default timeout for this operation is governed by :attr:`~.Cluster.max_schema_agreement_wait`
+        and :attr:`~.Cluster.control_connection_timeout`.
+        Passing schema_agreement_wait here overrides :attr:`~.Cluster.max_schema_agreement_wait`.
+        Setting schema_agreement_wait <= 0 will bypass schema agreement and refresh schema immediately.
+        RuntimeWarning is raised if schema refresh fails for any reason.
+        """
+        if not self.control_connection.refresh_schema(keyspace, table, usertype, schema_agreement_wait):
+            raise RuntimeWarning("Schema was not refreshed. See log for details.")
```

Comment:
I don't think it raises Exceptions. They are all caught and logged.
Regardless, good input on the warning. Updated.